src/sentry/static/sentry/app/components/splitDiff.tsx

```diff
@@ -0,0 +1,121 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from '@emotion/styled';
+import {diffChars, diffWords, diffLines, Change} from 'diff';
+
+const diffFnMap = {
+  chars: diffChars,
+  words: diffWords,
+  lines: diffLines,
+};
+
+type Props = {
+  base: string;
+  target: string;
+  type: keyof typeof diffFnMap;
+  className: string;
+};
+
+export function SplitDiff({className, type, base, target}: Props) {
```

Comment:
Please shouldn't this function be a functional component?

Reply:
Isn't it already a function component?

Reply:
Oh yes ...it's a functional component, but I meant...in this case shouldn't we use ES6 arrow function and assign it to a const as we usually do?